Tests/Tests/Core/Utilities/UserDefaults/UserDefaultTests.swift

```diff
@@ -34,7 +34,21 @@ class UserDefaultTests: XCTestCase {
     // When
     Defaults.isAuthenticated = givenValue
     // Then
-    XCTAssertEqual(userDefaults.bool(forKey: Defaults.testBoolKey), Defaults.isAuthenticated)
+    XCTAssertEqual(givenValue, Defaults.isAuthenticated)
+  }
+
+  func testResetValue() {
+    // Given
+    let givenValue = Defaults.testStringKey
+    
+    // Set an initial value
+    Defaults.screenName = givenValue
+    
+    // When
+    Defaults.$screenName.resetValue()
+    
+    // Then
+    XCTAssertEqual(Defaults.screenName, "default")
   }
```

Comment:
Could we please add additional tests that are covering other data types. It would be good to have an example for storing Codable data as well.